src/modules/keyboardmanager/common/KeyboardManagerState.cpp

```diff
@@ -191,6 +182,54 @@ bool KeyboardManagerState::AddAppSpecificShortcut(const std::wstring& app, const
     return true;
 }
 
+// Function to get the iterator of a single key remap given the source key. Returns nullopt if it isn't remapped
+std::optional<SingleKeyRemapTable::iterator> KeyboardManagerState::GetSingleKeyRemap(const DWORD& originalKey)
+{
+    auto it = singleKeyReMap.find(originalKey);
+    if (it != singleKeyReMap.end())
+    {
+        return it;
+    }
+
+    return std::nullopt;
+}
+
+bool KeyboardManagerState::CheckShortcutRemapInvoked(const std::optional<std::wstring>& appName)
+{
+    // Assumes appName exists in the app-specific remap table
+    ShortcutRemapTable& currentRemapTable = appName ? appSpecificShortcutReMap[*appName] : osLevelShortcutReMap;
+    for (auto& it : currentRemapTable)
+    {
+        if (it.second.isShortcutInvoked)
+        {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+std::vector<Shortcut>& KeyboardManagerState::GetSortedShortcutRemapVector(const std::optional<std::wstring>& appName)
+{
+    // Assumes appName exists in the app-specific remap table
+    return appName ? appSpecificShortcutReMapSortedKeys[*appName] : osLevelShortcutReMapSortedKeys;

```

Comment:
[clarification] What happens if the app name doesn't exist? Will this throw an error and is it handled?

Reply:
All these methods are called in the `HandleShortcutRemapEvent` method, which gets called only after checking if the `appName` is present in the remapping tables, which is why there is no code to check that. If I were to change this to an `if/else` block I would have to initialize the result to a new vector, which could add memory overhead. Not sure what is the better practice here. @ryanbodrug-microsoft thoughts on this usage of the ternary op?